File: Python/Algorithms/CircularArrayRotation.py
#Problem link: https://www.hackerrank.com/challenges/circular-array-rotation/
#This is the slow implementation

#!/bin/python3

import sys

def rotate(array, times):
    for i in range(0, times):
        left = array[0];
        for j in range (1, len(array)):
            swap = array[j];
            array[j] = left;
            left = swap;

        array[0] = left;


n, k, q = input().strip().split(' ')
n, k, q = [int(n),int(k),int(q)]
a = [int(a_temp) for a_temp in input().strip().split(' ')]

rotate(a, k)

for a0 in range(q):
    m = int(input().strip())
    print(a[m])

File: C#/Algorithms/Warmup/CircularArrayRotation.cs
using System;

namespace HackerRank
{
    /// <summary>
    /// Problem link: https://www.hackerrank.com/challenges/circular-array-rotation 
    /// This is the fast implementation.
    /// </summary>
    class CircularArrayRotation
    {
        public static void Main()
        {
            string[] tokens_n = Console.ReadLine().Split(' ');
            int k = Convert.ToInt32(tokens_n[1]);
            int q = Convert.ToInt32(tokens_n[2]);
            string[] a_temp = Console.ReadLine().Split(' ');
            int[] a = Array.ConvertAll(a_temp, Int32.Parse);

            var position = getPositionAfterRotaion(a, k);

            for (int a0 = 0; a0 < q; a0++)
            {
                int m = Convert.ToInt32(Console.ReadLine());
                var curPos = position;

                for (int j = 0; j < m; j++)
                {
                    if (curPos < a.Length - 1) curPos++;
                    else curPos = 0;
                }
                Console.WriteLine(a[curPos]);
            }
        }

        private static int getPositionAfterRotaion(int[] array, int times)
        {
            int pos = 0;
            for (int i = 0; i < times; i++)
            {
                if (pos > 0) pos--;
                else pos = array.Length - 1;
            }

            return pos;
        }
    }
}
